### PR TITLE
Export a11y-testcases

### PR DESCRIPTION
**Pull request by jkodu**
This PR generates the test snippets from every rule into an explicit format, for easy consumption in various tools like `axe-core`.

- the test cases are generated in `a11y-testcases` directory using the command `npm run build`.
- [a11y-testcases.zip](https://github.com/auto-wcag/auto-wcag/files/2349249/a11y-testcases.zip) download the zip file to get an idea of what gets generated.
- a zip is also generated, which will me made available for download from the auto-wcag website.

An example structure of json which has a listing of all the test cases can be seen here:

```
{
  "name": "auto-wcag",
  "license": "https://auto-wcag.github.io/auto-wcag/license.html",
  "description": "Test Cases for auto-wcag rules",
  "a11y-testcases": [
    {
      "url": "https://auto-wcag.github.io/auto-wcag/auto-wcag-testcases/assets/SC1-1-1-image-has-name_passed_example_1.html",
      "relativeUrl": "assets/SC1-1-1-image-has-name_passed_example_1.html",
      "successCriteria": [
        "WCAG2:non-text-content"
      ],
      "passed": [
        "*"
      ],
      "ruleId": "SC1-1-1-image-has-name",
      "rulePage": "https://auto-wcag.github.io/auto-wcag/rules/SC1-1-1-image-has-name.html"
    },
    {
      "url": "https://auto-wcag.github.io/auto-wcag/auto-wcag-testcases/assets/SC1-2-1-media-alternative-audio_failed_example_1.html",
      "relativeUrl": "assets/SC1-2-1-media-alternative-audio_failed_example_1.html",
      "successCriteria": [

      ],
      "failed": [
        "*[data-rule-target]"
      ],
      "ruleId": "SC1-2-1-media-alternative-audio",
      "rulePage": "https://auto-wcag.github.io/auto-wcag/rules/SC1-2-1-media-alternative-audio.html"
    },
...MORE TESTS...
```

Closes issue: 
- https://github.com/auto-wcag/auto-wcag/issues/184

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [ ] Make sure you requesting to **pull a issue/feature/bugfix branch** (right side) to the master branch (left side).
- [ ] Make sure the pull request is prefixed with `Rule:` or `Algorithm:` or `Chore:` based on work involved.

# How to Review And Approve
- Go to the “files changed” tab, there you will have the option to leave comments on different lines. 
- Once the review is completed, find the “Review changes” button in the top right, select “approve” and click “Submit review”.

### CODE REVIEW

**Comment by jkodu** [comment]
Note:
Below cases have to be covered before this can be merged.

- [ ] leaning towards implicit format for test cases that are exported (include url and id).
- [ ] each rule test cases need to be decorated with `data-target` attribute.

**Comment by jkodu** [comment]
@WilcoFiers - Updated, please review.

File: package.json [hunk 1]
```diff
@@ -2,13 +2,18 @@
...
   "version": "1.0.0",
   "description": "auto-wcag",
-  "main": "index.js",
+	"main": "index.js",
+	"site-url-prefix": "https://auto-wcag.github.io/auto-wcag",
 	"testcases-embeds-dir": "_testcases-embeds/",
-	"testcases-export-dir": "_testcases-export/",
+	"testcases-export-dir": "auto-wcag-testcases/",
+	"testcases-export-filename": "testcases.json",
+	"rules-export-filename": "rules.json",
```
Comment (kasperisager) [inline package.json]: Could all these configuration variables be put in a `config` object, which is the standard approach to npm package configuration? https://docs.npmjs.com/files/package.json#config

File: package.json [hunk 2]
```diff
@@ -2,13 +2,18 @@
...
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
 		"dev": "bundle exec jekyll serve",
+		"build": "bundle exec jekyll build",
 		"generate-wcag-data": "node wcag-url-generator",
+		"export-tests": "bundle exec jekyll build",
```
Comment (kasperisager) [inline package.json]: Why include two identical scripts?

File: package.json [hunk 3]
```diff
@@ -20,5 +25,5 @@
   "bugs": {
     "url": "https://github.com/auto-wcag/auto-wcag/issues"
   },
-  "homepage": "https://github.com/auto-wcag/auto-wcag#readme"
-}
+	"homepage": "https://github.com/auto-wcag/auto-wcag#readme"
+}
```
Comment (kasperisager) [inline package.json]: EOF newline removed; I'd suggest we add an `.editorconfig` to prevent these inconsistencies from sneaking in.

**Comment by jkodu** [comment]
@kasperisager, cheers for the review. Updated. Review again.

**Pull request merged**